File: src/Java/BehavioralPatterns/Mediator/ChainReaction/src/Mediator.java
public class Mediator {
    private Button button;
    private Dropdown dropdown;
    private TextBox textBox;
    private FontEditor fontEditor;

    public Mediator() {
        dropdown = new Dropdown(this);
        button = new Button(this);
        textBox = new TextBox(this);
        fontEditor = new FontEditor(this);
    }

    public void onStateChanged(Participant participant) {
        if (participant == textBox && textBox.isEnabled()) {
            fontEditor.enable();
            return;
        }

        if (participant == textBox && !textBox.isEnabled()) {
            fontEditor.disable();
            return;
        }

        if (participant == dropdown && dropdown.getSelectedItem().equals("Manual")) {
            button.enable();
            textBox.enable();
            return;
        }

        if (participant == dropdown && dropdown.getSelectedItem().equals("Auto")) {
            button.disable();
            textBox.disable();
        }
    }

    public void setButton(Button button) {
        this.button = button;
    }

    public void setDropdown(Dropdown dropdown) {
        this.dropdown = dropdown;
    }

    public void setTextBox(TextBox textBox) {
        this.textBox = textBox;
    }

    public void setFontEditor(FontEditor fontEditor) {
        this.fontEditor = fontEditor;
    }

    public Button getButton() {
        return button;
    }

    public Dropdown getDropdown() {
        return dropdown;
    }

    public TextBox getTextBox() {
        return textBox;
    }

    public FontEditor getFontEditor() {
        return fontEditor;
    }
}

File: src/PHP/BehavioralPatterns/Mediator/ChainReaction/Mediator.php
<?php

namespace BehavioralPatterns\Mediator\ChainReaction;

class Mediator {
    private Button $button;
    private Dropdown $dropdown;
    private TextBox $textBox;
    private FontEditor $fontEditor;

    public function __construct() {
        $this->dropdown = new Dropdown($this);
        $this->button = new Button($this);
        $this->textBox = new TextBox($this);
        $this->fontEditor = new FontEditor($this);
    }

    /**
     * @return Button
     */
    public function getButton(): Button
    {
        return $this->button;
    }

    /**
     * @return Dropdown
     */
    public function getDropdown(): Dropdown
    {
        return $this->dropdown;
    }

    /**
     * @return TextBox
     */
    public function getTextBox(): TextBox
    {
        return $this->textBox;
    }

    /**
     * @return FontEditor
     */
    public function getFontEditor(): FontEditor
    {
        return $this->fontEditor;
    }

    public function onStateChanged(Participant $participant): void
    {
        if ($participant === $this->textBox && $this->textBox->isEnabled()) {
            $this->fontEditor->enable();
            return;
        }

        if ($participant === $this->textBox && !$this->textBox->isEnabled()) {
            $this->fontEditor->disable();
            return;
        }

        if ($participant === $this->dropdown && $this->dropdown->getSelectedItem() == "Manual") {
            $this->button->enable();
            $this->textBox->enable();
            return;
        }

        if ($participant === $this->dropdown && $this->dropdown->getSelectedItem() == "Auto") {
            $this->button->disable();
            $this->textBox->disable();
        }
    }
}